### PR TITLE
fix: Fix DesignInfo mdx component handling of multiple design/code authors

### PR DESCRIPTION
**Pull request by HaasJona**
The variable `person` was undefined in the scope of the join method.

Related to #7168

### CODE REVIEW

File: sites/orgdocs/src/theme/MDXComponents/design-info.mjs [hunk 1]
```diff
@@ -200,9 +200,14 @@ export const DesignInfo = ({ design }) => {
                 {typeof designInfo[design].design === 'string' ? (
                   <span>{designInfo[design].design}</span>
                 ) : (
-                  designInfo[design].design
-                    .map((person) => <span key={person}>{person}</span>)
-                    .join(<span key={`${person}_join`}>, </span>)
+                  designInfo[design].design.map((person, index) => (
+                    <span key={person}>
+                      {person}
+                      {index < designInfo[design].design.length - 1 && (
```
Comment (joostdecock) [inline sites/orgdocs/src/theme/MDXComponents/design-info.mjs:206]: Can't we just replace:
```js
.join(<span key={`${person}_join`}>, </span>)
```
with:
```js
.join(<span>, </span>)
```

As the `join()` call turns the array in a string, we don't need a key, do we? (didn't try this)

File: sites/orgdocs/src/theme/MDXComponents/design-info.mjs [hunk 2]
```diff
@@ -217,9 +222,14 @@ export const DesignInfo = ({ design }) => {
                 {typeof designInfo[design].code === 'string' ? (
                   <span>{designInfo[design].code}</span>
                 ) : (
-                  designInfo[design].code
-                    .map((person) => <span key={person}>{person}</span>)
-                    .join(<span key={`${person}_join`}>, </span>)
+                  designInfo[design].map((person, index) => (
+                    <span key={person}>
+                      {person}
+                      {index < designInfo[design].design.length - 1 && (
```
Comment (joostdecock) [inline sites/orgdocs/src/theme/MDXComponents/design-info.mjs:228]: Same remark

**Review by joostdecock — COMMENTED** [review]
@HaasJona Can't we just drop the key here?

File: sites/orgdocs/src/theme/MDXComponents/design-info.mjs (hunk 1 above)
Comment (HaasJona) [reply sites/orgdocs/src/theme/MDXComponents/design-info.mjs:206]: I don't think we want a string, we want an array of react elements. Maybe we could build a string, but then we wouldn't need/want any `span`s at all, but then we also couldn't highlight or link the people in the future.

Comment (BenJamesBen) [reply sites/orgdocs/src/theme/MDXComponents/design-info.mjs:206]: I tried what I think the suggested alternative was, and it resulted in:
Screenshot 2024-10-23 at 6 00 44 AM

While the PR code changes produced a more correct result:
Screenshot 2024-10-23 at 6 01 26 AM

Comment (HaasJona) [reply sites/orgdocs/src/theme/MDXComponents/design-info.mjs:206]: That's about what I expected, as converting react elements to string just results in `[object Object]`.

What could work is something like

```js
                  designInfo[design].design.join(", ")
```

which means we wouldn't create any react elements at all and just build a string, but long-term that would prevent us from formatting the names differently than the commas (e.g. by linking to a profile page of that person).

Comment (joostdecock) [reply sites/orgdocs/src/theme/MDXComponents/design-info.mjs:206]: Alright, I got it wrong. Merging.

**Review by joostdecock — APPROVED** [review]

**Pull request merged**